esphome/components/online_image/bmp_image.cpp

```diff
@@ -101,6 +122,34 @@ int HOT BmpDecoder::decode(uint8_t *buffer, size_t size) {
       }
       break;
     }
+    case 8: {
+      while (index < size) {
+        uint8_t color_index = buffer[index];
+
+        if (color_index >= this->color_table_entries_) {
+          ESP_LOGE(TAG, "Invalid color index: %d", color_index);
+          continue;
```

Comment:
Do you really want to continue here? Or better just abort the decoding and return an error?
```suggestion
          return DECODE_ERROR_UNSUPPORTED_FORMAT;
```

Reply:
Its fine with me.